swift/Arrow/Sources/Arrow/ArrowDecoder.swift

```diff
@@ -0,0 +1,343 @@
+    }
+
+    func decode(_ type: UInt8.Type, forKey key: Key) throws -> UInt8 {
+        return try self.decoder.doDecode(key)!
+    }
+
+    func decode(_ type: UInt16.Type, forKey key: Key) throws -> UInt16 {
+        return try self.decoder.doDecode(key)!
+    }
+
+    func decode(_ type: UInt32.Type, forKey key: Key) throws -> UInt32 {
+        return try self.decoder.doDecode(key)!
+    }
+
+    func decode(_ type: UInt64.Type, forKey key: Key) throws -> UInt64 {
+        return try self.decoder.doDecode(key)!
+    }
+
+    func decode<T>(_ type: T.Type, forKey key: Key) throws -> T where T: Decodable {
+        if type == Date.self {
+            return try self.decoder.doDecode(key)!
+        } else {
+            throw ArrowError.invalid("Type \(type) is currently not supported")
+        }
+    }
+
+    func nestedContainer<NestedKey>(
+        keyedBy type: NestedKey.Type,
+        forKey key: Key
+    ) throws -> KeyedDecodingContainer<NestedKey> where NestedKey: CodingKey {
+        throw ArrowError.invalid("Nested decoding is currently not supported.")
+    }
+
+    func nestedUnkeyedContainer(forKey key: Key) throws -> UnkeyedDecodingContainer {
+        throw ArrowError.invalid("Nested decoding is currently not supported.")
+    }
+
+    func superDecoder() throws -> Decoder {
+        throw ArrowError.invalid("super decoding is currently not supported.")
+    }
+
+    func superDecoder(forKey key: Key) throws -> Decoder {
+        throw ArrowError.invalid("super decoding is currently not supported.")
+    }
+}
+
+private struct ArrowSingleValueDecoding: SingleValueDecodingContainer {
+    var codingPath = [CodingKey]()
+    let decoder: ArrowDecoder
+
+    init(_ decoder: ArrowDecoder, codingPath: [CodingKey]) {
+        self.decoder = decoder
+        self.codingPath = codingPath
+    }
+
+    func decodeNil() -> Bool {
+        return false
+    }
+
+    func decode(_ type: Bool.Type) throws -> Bool {
+        return try self.decoder.doDecode(0)!
+    }
+
+    func decode(_ type: String.Type) throws -> String {
+        return try self.decoder.doDecode(0)!
+    }
+
+    func decode(_ type: Double.Type) throws -> Double {
+        return try self.decoder.doDecode(0)!
+    }
+
+    func decode(_ type: Float.Type) throws -> Float {
+        return try self.decoder.doDecode(0)!
+    }
+
+    func decode(_ type: Int.Type) throws -> Int {
+        return try self.decoder.doDecode(0)!
+    }
+
+    func decode(_ type: Int8.Type) throws -> Int8 {
+        return try self.decoder.doDecode(0)!
+    }
+
+    func decode(_ type: Int16.Type) throws -> Int16 {
+        return try self.decoder.doDecode(0)!
+    }
+
+    func decode(_ type: Int32.Type) throws -> Int32 {
+        return try self.decoder.doDecode(0)!
+    }
+
+    func decode(_ type: Int64.Type) throws -> Int64 {
+        return try self.decoder.doDecode(0)!
+    }
+
+    func decode(_ type: UInt.Type) throws -> UInt {
+        return try self.decoder.doDecode(0)!
```

Comment:
We may want to use `throw ArrowError.invalid("UInt type is not supported (please use UInt8, UInt16, UInt32 or UInt64)")` here.

Reply:
Good catch! I will add the error.